### DISCUSSION TITLE
Bypass Lock Screen when launching apps from Fermata Mirror launcher

### DISCUSSION THREAD

**Discussion by plapic** [AndreyPavlenko/Fermata#519] [Announcements]
Version 1.99 implemented a launcher for the Fermata Mirror which is great. My question is it loads immediately when I select Fermata Mirror in AA however as soon as I select one the apps from the launcher it requires me to unlock the device and as I recall the previous version I had to unlock Fermata Mirror before it could run but now somehow that is bypassed so can you bypass the lock screen when launching apps?

**Comment by AndreyPavlenko** [comment]
With the following adb command you can disable the permission request:
```
adb shell appops set me.aap.fermata.auto.dear.google.why PROJECT_MEDIA allow
```
There is also an auto-confirmation of the screen cast request, but it does not work on some devices and the device unlock may be required.

**Reply by plapic** [reply]
I did try that and also did **adb shell appops get me.aap.fermata.auto.dear.google.why** to make sure it was set and it returned _PROJECT_MEDIA: allow; time=+5h1m32s901ms ago; rejectTime=+165d4h16m56s520ms ago_ but it still requests to unlock. Is this an known issue for Samsung S20+ running Android 13?
Is the Mirror part of me.aap.fermata.auto.dear.google.why or a separate app?

What did you mean _auto-confirmation of the screen cast request_?

**Reply by AndreyPavlenko** [reply]
This command grants the screen projections permission and you should not have the permission request anymore. However, there is also the screen cast confirmation dialog and the apps attempts to auto confirm this dialog. I've not tested on samsung devices, but afik it does not work.

**Reply by plapic** [reply]
I don't get screen cast confirmation just lock screen. Once I manually unlock it stays unlocked even when I select another app from the mirror launcher. So it seems the issue occurs only on the first launch of an app

**Reply by AndreyPavlenko** [reply]
The app doesn't have permissions to unlock the screen. The launcher and Fermata could work on a locked device, but not the other apps, that do not support this.
